src/App.tsx

```diff
@@ -1,73 +1,113 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
 import './App.scss';
+
 import { peopleFromServer } from './data/people';
 
+import { Person } from './types/Person';
+
+const peopleWithId: Person[] = peopleFromServer.map((person, index) => {
+  return {
+    ...person,
+    id: index + 1,
+  };
+});
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [currentPerson, setCurrentPerson] = useState<Person | null>(null);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  const { name, born, died } = currentPerson || {};
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
```

Comment:
Correct way to do this is:

```suggestion
  const applyQuery = useCallback(debounce(setAppliedQuery, 300), [setAppliedQuery]);
```